### PR TITLE
Migrate Appointments Controller

### PR DESCRIPTION
**Pull request by vkrmis**
**Story card:** https://app.clubhouse.io/simpledotorg/story/1133/migrate-the-appointmentscontroller-and-its-actions
## Because

We're preparing all controllers for new permissions so we can darklaunch to production

## This addresses

Migrates the AppointmentsController and it's views to use the new permissions (gated by a Flipper flag)

### CODE REVIEW

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -85,7 +85,7 @@
 
               <% if current_admin.accessible_facilities(:manage_overdue_list).any? %>
                 <li class="nav-item">
-                  <%= link_to "Overdue patients", appointments_path, class: "nav-link #{active_controller?("appointments")}" %>
+                  <%= link_to "Overdue", appointments_path, class: "nav-link #{active_controller?("appointments")}" %>
```
Comment (harimohanraj89) [inline app/views/layouts/application.html.erb:88]: Why this change?

**Review by harimohanraj89 — APPROVED** [review]
This looks super straightforward, nice. :shipit: :shipit: :shipit:

File: app/views/layouts/application.html.erb (hunk 1 above)
Comment (vkrmis) [reply app/views/layouts/application.html.erb:88]: accidentally included " patients" in an earlier PR, the old-permissions branch of the layout just says "Overdue" now, so I've removed the extra " patients" here.

Comment (harimohanraj89) [reply app/views/layouts/application.html.erb:88]: Noice 😎
